applications/Diffy_chat.md

```diff
@@ -0,0 +1,133 @@
+
+Dmitrii Shevchenko is a TechLead engineer with 10+ years of experience in developing and integrating IT, networking, security and blockchain solutions. Involved in implementation of highly reliable industrial solutions and development of FinTech and DeFi applications.
+
+Nikita Orlov, ETH Waterloo 2019 hackathon prize-winner, is a TechLead engineer with over 8 years of experience in development and integration of fault-tolerant high-loaded SaaS IT solutions including relevant experience in blockchain solutions.
+
+Alexander Plekhanov is a full stack software developer with over 5 years of experience including blockchain-based projects, enterprise solutions for fintech, call-centers, government services. Recent time mostly focused on smart-contracts development.
+
+Valeriy Tetevin is a programming engineer with over 8 years of experience in cloud-native applications. He also has strong knowledge of microservices architecture and back-end development for high-loaded applications.
+
+### Team Code Repos
+
+- https://github.com/1db1
+- https://github.com/StringNick
+- https://github.com/SealTV
+
+### Team LinkedIn Profiles (if available)
+
+- [Max Remov](https://www.linkedin.com/in/remov)
+- [Alexey Vexin](https://www.linkedin.com/in/alexey-vexin-800a4068/)
+- [Nikita Orlov](https://www.linkedin.com/in/nikita-orlov-aa6910186/)
+- [Alexander Plekhanov](https://www.linkedin.com/in/alexander-plekhanov/)
+- [Valeriy Tetevin](https://www.linkedin.com/in/sealtv/)
+
+## Development Roadmap :nut_and_bolt:
+
+We plan to execute 3 deliverables in two milestones:
+- a Substrate pallet for chat initiation;
+- a DOTRTC library for p2p transport implementation with test html pages for message passing;
+- a web-messenger dapp MVP with polkadot wallets authorisation.
+
+The project will be supported by a team of 2 developers, 1 UI/UX designer, 1 DevOps engineer and 1 QA.
+
+### Overview
+
+- **Total Estimated Duration:** 3 months
+- **Full-Time Equivalent (FTE):**  3,5 FTE
+- **Total Costs:** 30,000 USDT
+
+### Milestone 1 — Design and development of a pallet and a DOTRTC library
+
+- **Estimated duration:** 1,5 months
+- **FTE:**  2
+- **Costs:** 16,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how the developed features work. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure new functionality and robustness. In the guide, we will describe how to run these tests. |
+| 1. | Diffy chat pallet | We will develop a pallet (using Rust) which will include event handling to send and approve WebRTC offers for chat initiation. |
```

Comment:
What happens if a user is offline?

Reply:
Dear, @Noc2 , thank you for your quick response!
Let me answer step by step:
1. I suppose this is a translation issue) Offcourse, library won't do anything by itself, but it will include methods for the described functionality. On the first milestone we will develop this library and on the second milestone this library will be used by the user web dapp.
2. We will implement e2e encryption using the rs25519 algorithm (with Polkadot user keys) in the DOTRTC library to encrypt and decrypt packets.
3. No, we do not have a mock up, just a simple picture, that is in the application.
4. If a recipient user is offline the sender will receive a message, that he is offline and no connection will be set. A feature for sending messages to offline users is mentioned in our future plans.

Reply:
oh, I found a sentence "The library will.") Sorry, this was an edit error, I've just deleted it.